strawberryfields/tdm/tdmprogram.py

```diff
@@ -282,6 +310,121 @@ def context(self, *args, copies=1, shift="default"):
         self.loop_vars = self.params(*[f"p{i}" for i in range(len(args))])
         return self
 
+    def compile(self, *, device=None, compiler=None):
+        """Compile the time-domain program given a Strawberry Fields photonic hardware device specification.
+        At this stage the compilation is simply a check that the program matches the device.
+
+        Args:
+            device (~strawberryfields.api.DeviceSpec): device specification object to use for
+                program compilation
+            compiler (str, ~strawberryfields.compilers.Compiler): Compiler name or compile strategy
+                to use. If a device is specified, this overrides the compile strategy specified by
+                the hardware :class:`~.DeviceSpec`. For :class:`~.TDMProgram` this should be set to
+                None.
```

Comment:
This needs to be updated 🙂 Since we can now pass `gaussian`, `tdm_one_loop`, etc.